src/arch/southern-islands/timing/Gpu.cc

```diff
@@ -134,6 +134,21 @@ void Gpu::MapNDRange(NDRange *ndrange)
 }
 
 
+void Gpu::UnmapNDRange(NDRange* ndrange)
+{
+	// Erase every workgroup in each compute unit, setting the
+	// work_groups size to 0
+	for (int i = 0; i < num_compute_units; i++)
```

Comment:
You can use 2 lines of code for this:

    for (auto &compute_unit: compute_units)
        compute_unit->Reset();